gdal/frmts/grib/gribdataset.cpp

```diff
@@ -981,10 +1089,123 @@ void GRIBRasterBand::UncacheData()
 
 GRIBRasterBand::~GRIBRasterBand()
 {
-    CPLFree(longFstLevel);
+    if (longFstLevel != nullptr)
+        CPLFree(longFstLevel);
     UncacheData();
 }
 
+
+/************************************************************************/
+/*                           InventoryWrapperGrib                       */
+/************************************************************************/
+class InventoryWrapperGrib : public gdal::grib::InventoryWrapper
+{
+  public:
+    explicit InventoryWrapperGrib(VSILFILE *fp) : gdal::grib::InventoryWrapper()
+    {
+        result_ = GRIB2Inventory(fp, &inv_, &inv_len_, 0 /* all messages */, &num_messages_);
+    }
+
+    ~InventoryWrapperGrib() override
+    {
+        if (inv_ == nullptr)
+            return;
+        for (uInt4 i = 0; i < inv_len_; i++)
+        {
+            GRIB2InventoryFree(inv_ + i);
+        }
+        free(inv_);
+    }
+};
+
+/************************************************************************/
+/*                           InventoryWrapperSidecar                    */
+/************************************************************************/
+
+class InventoryWrapperSidecar : public gdal::grib::InventoryWrapper
+{
+  public:
+    explicit InventoryWrapperSidecar(VSILFILE *fp) : gdal::grib::InventoryWrapper()
+    {
+        result_ = -1;
+        VSIFSeekL(fp, 0, SEEK_END);
+        size_t length = static_cast<size_t>(VSIFTellL(fp));
+        if (length > 4*1024*1024) return;
+        std::vector<char> psSidecar(static_cast<size_t>(length));
```

Comment:
useless cast, and naming convention fix:
```suggestion
        std::vector<char> achSidecar(length);
```